packages/create-sitecore-jss/src/templates/nextjs-sxa/src/lib/page-props-factory/plugins/error-pages.ts

```diff
@@ -0,0 +1,44 @@
+import { GetServerSidePropsContext, GetStaticPropsContext } from 'next';
+import { SitecorePageProps } from 'lib/page-props';
+import { Plugin } from 'lib/page-props-factory';
+import { GraphQLErrorPagesService } from '@sitecore-jss/sitecore-jss-nextjs';
+import { isServerSidePropsContext } from '..';
+import config from 'temp/config';
+
+class ErrorPagesPlugin implements Plugin {
+  order = 2;
+
+  async exec(props: SitecorePageProps, context: GetServerSidePropsContext | GetStaticPropsContext) {
+    const errorPagesService = new GraphQLErrorPagesService({
+      endpoint: config.graphQLEndpoint,
+      apiKey: config.sitecoreApiKey,
+      siteName: config.jssAppName,
+      language: props.locale,
+    });
+    const resultErrorPages = await errorPagesService.fetchErrorPages();
```

Comment:
We should only fetch these if there is a reason to (for performance reasons). Before this, we should first check for error state (i.e. notFound or 5xx), and exit if all good.

Reply:
Yes, I agreed